internal/telemetry/ticker.go

```diff
@@ -0,0 +1,40 @@
+package telemetry
+
+import (
+	"math/rand"
+	"time"
+)
+
+const (
+	jitterUpperBound = 60 // s
+)
+
+// ticker represents a ticker interface that
+// can be mocked for more reliable tests.
+type ticker interface {
+	C() <-chan time.Time
+}
+
```

Comment:
I'm really missing Stop() here.

Reply:
(in the interface, I mean)

Reply:
That is right, added it in [7fe148b](https://github.com/grafana/synthetic-monitoring-agent/pull/621/commits/7fe148b383301d34e30a47fcd0b48a159d411c00).